engines/wage/detection.cpp

```diff
@@ -0,0 +1,240 @@
+		"",
+		AD_ENTRY1s("Camp Cantitoe", "098aa5c11c58e1ef274a30a9e01b4755", 621440),
+		Common::EN_ANY,
+		Common::kPlatformMacintosh,
+		ADGF_DEFAULT,
+		GUIO0()
+	},
+	{
+		"wage",
+		"Deep Angst",
+		AD_ENTRY1s("Deep Angst", "635f62bbc569e72b03cab9107927d03d", 335232),
+		Common::EN_ANY,
+		Common::kPlatformMacintosh,
+		ADGF_GENERIC,
+		GUIO0()
+	},
+	{
+		"wage",
+		"Escape from School!",
+		AD_ENTRY1s("Escape from School!", "a854be48d4af20126d18a9cad93a969b", 51840),
+		Common::EN_ANY,
+		Common::kPlatformMacintosh,
+		ADGF_GENERIC,
+		GUIO0()
+	},
+	{
+		"wage",
+		"Queen Quest",
+		AD_ENTRY1s("Queen Quest", "730605d312efedb5e3ff108522fcac18", 59776),
+		Common::EN_ANY,
+		Common::kPlatformMacintosh,
+		ADGF_GENERIC,
+		GUIO0()
+	},
+	{
+		"scepters",
+		"",
+		AD_ENTRY1s("Scepters", "b80bff315897776dda7689cdf829fab4", 360832),
+		Common::EN_ANY,
+		Common::kPlatformMacintosh,
+		ADGF_DEFAULT,
+		GUIO0()
+	},
+	{
+		"wage",
+		"ZikTuria",
+		AD_ENTRY1s("ZikTuria", "e793155bed1a70fa2074a3fcd696b751", 54784),
+		Common::EN_ANY,
+		Common::kPlatformMacintosh,
+		ADGF_GENERIC,
+		GUIO0()
+	},
+	{
+		"wage",
+		"Zoony",
+		AD_ENTRY1s("Zoony", "e6cc8a914a4215dafbcce6315dd12cf5", 160256),
+		Common::EN_ANY,
+		Common::kPlatformMacintosh,
+		ADGF_GENERIC,
+		GUIO0()
+	},
+
+	AD_TABLE_END_MARKER
+};
+
+} // End of namespace Wage
+
+class WageMetaEngine : public AdvancedMetaEngine {
+public:
+	WageMetaEngine() : AdvancedMetaEngine(Wage::gameDescriptions, sizeof(ADGameDescription), wageGames) {
+		_singleid = "wage";
+		_guioptions = GUIO2(GUIO_NOSPEECH, GUIO_NOMIDI);
+	}
+
+	virtual const char *getName() const {
+		return "World Adventure Game Engine";
+	}
+
+	virtual const char *getOriginalCopyright() const {
+		return "World Builder (C) Silicon Beach Software";
+	}
+
+	virtual bool createInstance(OSystem *syst, Engine **engine, const ADGameDescription *desc) const;
+	virtual bool hasFeature(MetaEngineFeature f) const;
+	virtual SaveStateList listSaves(const char *target) const;
+	virtual int getMaximumSaveSlot() const;
+	virtual void removeSaveState(const char *target, int slot) const;
+};
+
+bool WageMetaEngine::hasFeature(MetaEngineFeature f) const {
+	return
+		(f == kSupportsListSaves) ||
+		(f == kSupportsLoadingDuringStartup) ||
+		(f == kSupportsDeleteSave);
+}
+
+bool Wage::WageEngine::hasFeature(EngineFeature f) const {
+	return
+		(f == kSupportsRTL) ||
+		(f == kSupportsLoadingDuringRuntime) ||
+		(f == kSupportsSavingDuringRuntime);
+}
+
+bool WageMetaEngine::createInstance(OSystem *syst, Engine **engine, const ADGameDescription *desc) const {
+	if (desc) {
+		*engine = new Wage::WageEngine(syst, desc);
+	}
+	return desc != 0;
+}
+
+SaveStateList WageMetaEngine::listSaves(const char *target) const {
+	const uint32 WAGEflag = MKTAG('W','A','G','E');
+	Common::SaveFileManager *saveFileMan = g_system->getSavefileManager();
+	Common::StringArray filenames;
+	char saveDesc[31];
+	Common::String pattern = target;
+	pattern += ".???";
```

Comment:
06641f29a7bdcda280b0291f215193f055c83969

I suppose it was added after you branched for wage, so you'll probably have to merge in a recent master for this.